README.md

```diff
@@ -502,6 +502,47 @@ export default Base.extend({
 });
 ```
 
+### Upgrading to 2.0
+
+Starting from Ember Simple Auth 2.0, authorizers and the session service's `authorize` will be deprecated. You will need to configure your data adapter to include authorization data in request headers. Refer to the [Ember Guides](https://guides.emberjs.com/v3.0.0/models/customizing-adapters/#toc_headers-customization) for details on header configuration.
+
+The idea is to look up authorization data in the session service and include that data in your data adapter headers. Examples:
+
+```js
+// OAuth 2
+export default DS.JSONAPIAdapter.extend({
+  session: service('session'),
+  headers: computed('session.data.authenticated.access_token', function() {
+    let { access_token } = this.get('session.data.authenticated');
+    return {
+      Authorization: `Bearer ${access_token}`,
+    };
+  }),
+});
+```
+
+```js
+// Devise
+export default DS.JSONAPIAdapter.extend({
+  session: service('session'),
+  // defaults
+  // identificationAttributeName: 'email'
+  // tokenAttributeName: 'token'
+  headers: computed(
+    'session.data.authenticated.email',
+    'session.data.authenticated.token',
+    function() {
+      let { email, token } = this.get('session.data.authenticated');
+      let authData = `token="${token}", email="${email}"`;
+
+      return {
+        Authorization: `Token ${authData}`,
+      };
+    }
+  ),
+});
+```
```

Comment:
Have you checked whether these examples actually work yet? I'm not sure they do and we might have to add a new method to the `DataAdapterMixin`.

Reply:
@marcoow The examples don't work 🙂.

What method do you want to add to `DataAdapterMixin`? This works.
```js
// data-adapter-mixin.js
ajaxOptions() {
    let hash = this._super(...arguments);
    let { beforeSend } = hash;
    hash.beforeSend = (xhr) => {
    xhr.setRequestHeader('Authorization', this.get('authHeaderValue'));
    if (beforeSend) {
    ...
}
```
```js
// adapters/application.js
authHeaderValue: computed('session.data.authenticated.access_token', function() {
    let { access_token } = this.get('session.data.authenticated');
    return `Bearer ${access_token}`;
}),
```

Reply:
I think the best thing to do for now is to add a generic `authorize` method that receives the request as an argument:

```js
// data-adapter-mixin.js
ajaxOptions() {
  let hash = this._super(...arguments);
  let { beforeSend } = hash;
  hash.beforeSend = (xhr) => {
    this.authorize(xhr);
    if (beforeSend) {
    ...
}
```

```js
// adapters/application.js
authHeaderValue: computed('session.data.authenticated.access_token', function() {
  let { access_token } = this.get('session.data.authenticated');
  xhr.setRequestHeader('Authorization', `Bearer ${access_token}`);
}),
```

That way we'd allow multiple headers, passing the token via the query string etc.